### ISSUE TITLE
Problem pushing >1GB image with reverse proxy setup

### ISSUE THREAD

**Issue by Ataraxxia** [goharbor/harbor#16623]
Hello,

I am a new user of Harbor and docker registry in general. The problem i encountered is uploading an image that's 1,8GiB in size. Smaller ones (like 80MB  debian slim for example) work no problem, but this one stops uploading at around ~360MiB mark. The image composes of single layer that's it's full size. I have no means of modifying the image:

```
root@host1:~# docker push harbor.domain.org/scinet/ceos:4.27.3F
The push refers to repository [harbor.domain.org/scinet/ceos]
b69311ac82e4: Retrying in 1 second
```
It tries for a few times, then it gives up.

I am using harbor in http mode with separate vm with nginx acting as reverse proxy. 

The logs don't contain any error message.

The reverse proxy configuration:

```
server {
	listen 80;
	server_name harbor.domain.org;


        if ($scheme = http) {
        	return 301 https://$server_name$request_uri;
        }
}

server {
	listen 443 ssl http2;
	server_name harbor.domain.org;
    	
        ssl_certificate /etc/nginx/ssl/harbor.crt;
        ssl_certificate_key /etc/nginx/ssl/harbor.key;

        include /etc/nginx/snippets/ssl.conf;

        location / {
            proxy_pass http://172.24.114.12:80/;
            proxy_set_header Host $http_host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Proto $scheme;

        }
}
```

**Comment by Ataraxxia** [comment]
Unfortunately this does not help, espescially setting the proxy_buffer_size (I assume this is what you meant by proxy-body-size) to 0 completely breaks page display:

```
server {
	listen 443 ssl http2;
	server_name harbor.scinet.icm.edu.pl;
    	
        ssl_certificate /etc/nginx/ssl/harbor.crt;
        ssl_certificate_key /etc/nginx/ssl/harbor.key;

        include /etc/nginx/snippets/ssl.conf;

        location / {
            proxy_pass http://172.24.114.12:80/;
	    proxy_http_version 1.1;
            proxy_set_header Host $http_host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Proto $scheme;

	    proxy_buffering off;
	    proxy_request_buffering off;
            #proxy_buffer_size 0;
        }
}
```

Meanwhile I've tried to setup harbor directly, with it's own public IP and SSL and then the issue does not occur, but at this point I am curious as to why reverse proxying is being such a trouble

**Comment by Vad1mo** [comment]
because reverse proxying does it jobs and mingles all requests.

if this causes issues then try to set it only certain paths and not on the webui. 

```
/api/
/service/
/v2
/chartrepo/
/c/
```

**Comment by Ataraxxia** [comment]
It truned out that running reverse proxy is problematic when using nginx docker image included with install.sh.

tl;dr my solution: either use included nginx image or don't use it at all
